control_plane/src/local_env.rs

```diff
@@ -401,15 +401,18 @@ impl LocalEnv {
 
         fs::create_dir(base_path)?;
 
-        // generate keys for jwt
-        // openssl genrsa -out private_key.pem 2048
+        // Generate keys for jwt
         let private_key_path;
         if self.private_key_path == PathBuf::new() {
             private_key_path = base_path.join("auth_private_key.pem");
+
+            // Generate the public/private key pair
+            //
+            // openssl genpkey -algorithm ed25519 -out auth_private_key.pem
```

Comment:
This command doesn't work on macOS:
```
arthur@Arthurs-MacBook-Pro neon % openssl genpkey -algorithm ed25519 -out auth_private_key.pem
Algorithm ed25519 not found
usage: genpkey [-algorithm alg] [cipher] [-genparam] [-out file]
    [-outform der | pem] [-paramfile file] [-pass arg]
    [-pkeyopt opt:value] [-text]

 -algorithm name    Public key algorithm to use (must precede -pkeyopt)
 -genparam          Generate a set of parameters instead of a private key
 -out file          Output file to write to (default stdout)
 -outform format    Output format (DER or PEM)
 -paramfile file    File to load public key algorithm parameters from
                    (must precede -pkeyopt)
 -pass arg          Output file password source
 -pkeyopt opt:value Set public key algorithm option to the given value
 -text              Print the private/public key in human readable form
```

openssl version gives this:
```
arthur@Arthurs-MacBook-Pro neon % openssl version
LibreSSL 3.3.6
```

Perhaps it's possible to install another version of openssl, I'll try to find it.

Reply:
Hmm, I think we could use the `ring` crate directly to generate the key pair. That's what `jsonwebtoken` depends on anyway. I can take a look into that.

Reply:
[We have](https://github.com/neondatabase/neon#installing-dependencies-on-osx-1231) `brew install openssl` in our `README.md`, which installs a good version of openssl, but it's not added to PATH by default. If you add it to the PATH with:
```
echo 'export PATH="/usr/local/opt/openssl@3/bin:$PATH"' >> ~/.zshrc
```

Then `ed25519` key generation will work. We can ask to change PATH in our `README.md` or try to use `ring` crate. Later seems a better option.

Reply:
Unfortunately I could not make the key generation with `ring` crate work. It can generate the key pair, but I didn't find a way to output it to a PEM file that would be compatible with what the `jsonwebtoken` crate expects. If I understood correctly, `ring` stores the private key in PKCS#8 v2 format, while 'openssl' only supports PKCS#8 v1, and that's also what `jsonwebtoken` expects. I couldn't find a way to convert it.

So I suggest we advise changing the PATH. I don't have a mac; can you help to write the doc changes?

Reply:
Ok, I'll create a PR with readme fix, can be done independently to merging this PR.

Reply:
Can we use `ssh-keygen` to generate the keys?
https://github.com/neondatabase/neon/pull/3815#pullrequestreview-1339548138

Reply:
Is it possible to, on macs, just use the fully qualified path for the proper openssl version, to avoid needing to muck with the path?

Reply:
> Can we use ssh-keygen to generate the keys?

No, `ssh-keygen` uses yet another file format. I googled around and found some scripts to convert between the openssh and openssl formats but I couldn't get them to work either.

I also googled around for a python library that could generate the keys; this key generation is only used by the python tests. But I didn't find any good python solution either.

Reply:
> Ok, I'll create a PR with readme fix, can be done independently to merging this PR.

Thanks! Can you just add it to this PR, please? It'd be nice to keep all the changes in commit. We're not in a hurry to push this.

Reply:
Oh, I saw your PR for the README now. Thanks, i'll add it here!

Reply:
I think we're not finished with the docs PR yet, if that's really about the Python tests only, I would propose to actually add this path logic into Python.

But you've merged that already..

Reply:
Hmm, we do generate the keys, whenever you run `neon_local init`. But it's not really needed unless you enable authentication. It is disabled by default. So we could move the initialization to the python side.

Reply:
I found yet another solution: the `rcgen` crate. We were already using it in the proxy unit tests, to generate self-signed TLS certs.

Unfortunately rcgen  uses the PKCS#8 V2 format, same as `ring`, which is incompatible with openssl. But that's actually OK, 'jsonwebtoken' accepts both formats. So I think we can use it in the tests.

Reply:
I added that as a separate commit here. Please take a look.

Reply:
Ok, force-pushed yet another attempt at this. I ditched the 'rcgen' approach, and went back to using openssl. But I made the initialization code more lenient: if the 'openssl' command fails, and authentication was not enabled, it merely prints a message to the console and continues without the key pair:
```
$ ./target/debug/neon_local init
Could not generate keypair for JWT authentication: openssl failed: ''
Continuing anyway because authentication was not enabled
Initializing pageserver node 1 at '127.0.0.1:64000' in ".neon"
```
That works as long as you don't manually enable authentication in the config file after the initialization. That allows macOS users to continue without changing $PATH, unless you try to specifically work on authentication or run the authentication tests. I think that's nice for a casual developer who just wants to try it out; one less roadblock to stumble upon.